images/publish_image.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash -xe
+
+# Usage:
+#    ./publish_image.sh [OPTIONS] BUILD_TARGET REGISTRY REGISTRY_ORG
+#
+#    Build and publish infra images.
+#    If the -d flag is specified, will only build.
+
+main() {
+    local build_only
+    while getopts "d" opt; do
+        case "$opt" in
+            d)
+                build_only=true
+                ;;
+            *)
+                echo "Invalid argument: $opt"
+                exit 1
+        esac
+    done
+    shift $((OPTIND-1))
+    local build_target="${1:?}"
+    local registry="${2:?}"
+    local registry_org="${3:?}"
+    local full_image_name image_tag
+
+    image_tag="$(get_image_tag)"
+    full_image_name="$(
+        get_full_image_name \
+            "$registry" \
+            "$registry_org" \
+            "${build_target##*/}" \
+            "$image_tag"
+    )"
+
+    (
+        cd "$build_target"
+
+        build_image "$full_image_name"
+    )
+    [[ $build_only ]] && return
+    publish_image "$full_image_name"
+}
+
+get_image_tag() {
+    local current_commit
+    current_commit="$(git rev-parse HEAD)"
+    echo "${current_commit:0:6}"
```

Comment:
Can we also add the date the same way like it is done on the test-infra bootstrap image?

Reply:
The problem that I see with adding dates to tags is that you can build twice and get different builds from the same commit. Is this something that we want?

Reply:
Yes, because we don't have reproducible builds. It just reflects reality. It may or may not be different and therefore it should **not** look like the last build and should clearly be distinctable.